examples/pl_basics/profiler_example.py

```diff
@@ -34,14 +34,15 @@
 from lightning.pytorch.profilers.pytorch import PyTorchProfiler
 from lightning.pytorch.utilities.model_helpers import get_torchvision_model
 
+NUM_CUDA_DEVICES = torch.cuda.device_count()
 DEFAULT_CMD_LINE = (
```

Comment:
Does anyone know why we have this complicated dict handling at the top here? can't we just add these items as defaults to the LightningCLI(trainer_defaults=) dict?

Reply:
Probably for legacy reasons. I agree with moving these

Reply:
Sounds good, refactored this